### PR TITLE
[fastdds] Remove the need for super simple patches to be generated and removed …

### PR DESCRIPTION
**Pull request by KristianJerpetjon**
…the not needed boost dependency removed since fastdds 1.4.0

Specify library name and version:  **lib/1.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by KristianJerpetjon** [comment]
> TOPICS

Neither bumps to latest release neither removes a fake dependency

File: recipes/fast-dds/all/conanfile.py [hunk 1]
```diff
@@ -147,7 +146,17 @@ def validate(self):
                 raise ConanInvalidConfiguration("Mixing a dll {} library with a static runtime is a bad idea".format(self.name))
 
     def build(self):
-        self._patch_sources()
```
Comment (SpaceIm) [inline recipes/fast-dds/all/conanfile.py:150]: would be nice to remove the method as well, it's just dead code now.

File: recipes/fast-dds/all/conanfile.py [hunk 2]
```diff
@@ -114,9 +114,8 @@ def _configure_cmake(self):
     def requirements(self):
         self.requires("tinyxml2/9.0.0")
         self.requires("asio/1.21.0")
-        self.requires("fast-cdr/1.0.22")
+        self.requires("fast-cdr/1.0.23")
         self.requires("foonathan-memory/0.7.1")
-        self.requires("boost/1.75.0")  # boost/1.76 is required by version 2.3.2, boost/1.75.0 required for 2.3.3 by Windows
```
Comment (SpaceIm) [inline recipes/fast-dds/all/conanfile.py:119]: so it was a mistake?

Comment (KristianJerpetjon) [reply recipes/fast-dds/all/conanfile.py:119]: Well look at the fast-dds docs they say the dependency on boost was removed in 1.4.0 (1 major version ago) And it runs on my cross compiled target that doesn't have boost. So probably someone pushed some very old recipes to cci ?

File: recipes/fast-dds/all/conanfile.py (hunk 1 above)
Comment (KristianJerpetjon) [reply recipes/fast-dds/all/conanfile.py:150]: Agree pushed an update

File: recipes/fast-dds/all/conanfile.py (hunk 2 above)
Comment (SpaceIm) [reply recipes/fast-dds/all/conanfile.py:119]: but there is still 2.3.2 & 2.3.3 in this recipe, correct?

Comment (KristianJerpetjon) [reply recipes/fast-dds/all/conanfile.py:119]: https://fast-dds.docs.eprosima.com/en/latest/notes/notes.html#version-1-4-0 would be what you need to reach the same conclusion

Comment (KristianJerpetjon) [reply recipes/fast-dds/all/conanfile.py:119]: > but there is still 2.3.2 & 2.3.3 in this recipe, correct?

Yes  2.3.2-2.3.3 and 2.3.4 are still here but they are all newer than 1.4.0 so it should be fine

Comment (KristianJerpetjon) [reply recipes/fast-dds/all/conanfile.py:119]: seems 2.5.0 is to modern for gcc 5 any way to deal with that ?

Comment (KristianJerpetjon) [reply recipes/fast-dds/all/conanfile.py:119]: > seems 2.5.0 is to modern for gcc 5 any way to deal with that ?

Updating "validate" to reflect what is needed

File: recipes/fast-dds/all/conanfile.py [hunk 3]
```diff
@@ -144,10 +140,20 @@ def validate(self):
...
 
     def build(self):
-        self._patch_sources()
+        # dynamic patch works for 2.3.2 to 2.5.0
+        tools.replace_in_file("{}/CMakeLists.txt".format(self._source_subfolder),
+                              "eprosima_find_thirdparty(TinyXML2 tinyxml2)",
+                              "eprosima_find_thirdparty(tinyxml2 REQUIRED)")
+        tools.replace_in_file("{}/CMakeLists.txt".format(self._source_subfolder),
+                              "eprosima_find_thirdparty(Asio asio VERSION 1.10.8)",
+                              "eprosima_find_thirdparty(asio REQUIRED)")
```
Comment (alsora) [inline recipes/fast-dds/all/conanfile.py:153]: There's currently an error in the Fast-DDS patches.
See the code for `eprosima_find_thirdparty` here: https://github.com/eProsima/Fast-DDS/blob/master/cmake/common/eprosima_libraries.cmake#L168

The second argument shouldn't be `REQUIRED` but rather the name of the library.
It's not really important considering how we are using it here, but the patches should be changed into
```
eprosima_find_thirdparty(tinyxml2 tinyxml2)
eprosima_find_thirdparty(asio asio)
```

**Comment by SSE4** [comment]
@KristianJerpetjon there are conflicts, please rebase

**Pull request closed without merging**